pkg/porter/reconcile.go

```diff
@@ -122,12 +129,40 @@ func (p *Porter) ReconcileInstallation(ctx context.Context, opts ReconcileOption
 func (p *Porter) IsInstallationInSync(ctx context.Context, i claims.Installation, lastRun *claims.Run, action BundleAction) (bool, error) {
 	log := tracing.LoggerFromContext(ctx)
 
-	// Have we successfully completed the install action?
-	if !i.Status.InstallationCompleted || lastRun == nil {
-		log.Info("Triggering because the installation has not completed successfully yet")
+	// Only print out info messages if we are triggering a bundle run. Otherwise, keep the explanations in debug output.
+
+	// Has it been uninstalled? If so, we don't ever reconcile it again
+	if i.IsUninstalled() {
+		log.Info("Ignoring because the installation is uninstalled")
+		return true, nil
+	}
+
+	if i.Active {
+		// Should we install it?
+		if !i.IsInstalled() {
+			log.Info("Triggering because the installation has not completed successfully yet")
+			return false, nil
+		}
+	} else {
+		// Should we uninstall it?
+		if i.IsInstalled() {
+			log.Info("Triggering because the installation is now inactive")
+			return false, nil
+		}
+
+		// Otherwise ignore inactive
+		log.Info("Ignoring because the installation is inactive")
+		return true, nil
+	}
+
+	// We want to upgrade but we don't have values to compare against
+	// This shouldn't happen but check just in case
+	if lastRun == nil {
```

Comment:
Maybe I'm missing something here but I can't find a situation that this code path being executed

Reply:
That's why my comment included "This shouldn't happen but check just in case". 😀  It's just a quick check to prevent a panic in case our assumptions about how lastRun is passed changes.